### DISCUSSION TITLE
Listing IP addresses with site names

### DISCUSSION THREAD

**Discussion by markkuleinio** [netbox-community/netbox#11603] [Q&A]
I was faced with a need to list IP addresses + descriptions along with the site names, in the GUI.

As it currently is, there is no relation between IP addresses and sites. My first attempt to tackle this is a **script**:

1. Use `ObjectVar` to get the tenant that the user is interested in
2. Fetch all IP addresses of that tenant: `ip_addresses = IPAddress.objects.filter(tenant=data["tenant"])`)
3. Iterate over the IP addresses, find the associated prefix for every IP address: `prefix = Prefix.objects.get(prefix=str(ip.address.cidr))`
4. `self.log_info()` to output markdown with text (and links) showing the site name and IP address + description

This works as rough output, gets me numbered lines with whatever text (markdown) I want.

Is there a more clever way to output this as "NetBox-like" list of objects, with sorting and maybe filtering? Could a plugin do that?

I have plenty of experience using NetBox API but not these other ways of dealing with NetBox data (scripts, reports, plugins). I have always said that user input and data presentation are the hardest in everything 😄

**Comment by AMSG22** [comment]
have you figured this out?
I am in the same boat as you.
